### PR TITLE
fix(amplify-util-mock): non-promise lambda failing

### LINKED ISSUE #4019: AppSync Lambda mock is broken
**Describe the bug**

Using the `callback()` function in lambdas no longer works with the AppSync simulator.

Example:
````js
export default function (event, context, callback) {
  setTimeout(() => {
    callback(null, "response");
  }, 1000);
}
````

The response is always null.

The issue seems to have been introduced [here](https://github.com/aws-amplify/amplify-cli/pull/3254/files#diff-83f15b92fdd2f8b2d0ce1db08fd96fd9R83)

The reason is that the above function returns undefined, and `context.done(null, null);` is called immediately. When `callback()´ is finally called, it is already too late.

**Amplify CLI Version**
Starting at `amplify-util-mock@3.13.2`

**To Reproduce**
Use the example lambda above with the AppSync simulator.

**Expected behavior**
`invokeFunction()` should wait for the lambda to call the callback.

**Additional context**
Introduced in #3254 

cc: @ammarkarachi

### PR DESCRIPTION
**Pull request by bboure**
Fixes #4019

Non-async Lambda handlers should only rely on the value returned by the callback function.

See https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bboure** [comment]
@ammarkarachi This PR fixes #4019 and I think it should not affect #2502 
Please review

File: packages/amplify-nodejs-function-runtime-provider/src/__tests__/utils/__snapshots__/execute.test.ts.snap [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`execute async handler should fail when throwing an error 1`] = `Object {}`;
```
Comment (bboure) [inline packages/amplify-nodejs-function-runtime-provider/src/__tests__/utils/__snapshots__/execute.test.ts.snap:3]: @edwardfoyle I found a possible issue here.
As you can see the result is `Object {}`. I discovered that this is due to the `_.assign` [here](https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-nodejs-function-runtime-provider/src/utils/execute.ts#L40)

I guess the goal is to clone the object here?

Comment (edwardfoyle) [reply packages/amplify-nodejs-function-runtime-provider/src/__tests__/utils/__snapshots__/execute.test.ts.snap:3]: To be honest, I'm not sure what the intent of assigning to an object there was, but should be fine to leave it

**Comment by ChrisSargent** [comment]
Any timescale on when this will be released please?

**Comment by bboure** [comment]
@ChrisSargent This has been released already.
It should be available since version 1.0.3 of `amplify-nodejs-function-runtime-provider` and 4.19.0 of `amplify-cli` if I am not mistaken

**Comment by ChrisSargent** [comment]
@bboure ahhh, thanks. I didn't spot it in the changelog of the `@aws-amplify/cli` and because the original issue still had the `release pending` tag, I thought it wasn't (and didn't want to overwrite my local fix). Thanks!

**Pull request merged**